proc/variables.go

```diff
@@ -91,24 +101,40 @@ type EvalScope struct {
 	CFA    int64
 }
 
-func newVariable(name string, addr uintptr, dwarfType dwarf.Type, thread *Thread) (*Variable, error) {
+func newVariable(name string, addr uintptr, dwarfType dwarf.Type, thread *Thread) *Variable {
 	v := &Variable{
 		Name:      name,
 		Addr:      addr,
-		dwarfType: dwarfType,
+		DwarfType: dwarfType,
 		thread:    thread,
-		Type:      dwarfType.String(),
 	}
 
-	switch t := dwarfType.(type) {
+	v.RealType = v.DwarfType
+	for {
+		if tt, ok := v.RealType.(*dwarf.TypedefType); ok {
```

Comment:
It resolves typedefs. For example:

```
type notInt int

func main() {
    var i notInt
}
```

`i`'s `DwarfType` will be a `*dwarf.TypedefType` named `notInt`, after that loop `RealType` will be `int`. It's not an infinite loop because `tt.Type` is the type named by the typedef and:

```
type not legal
type legal not
```

does not compile.